rskj-core/src/main/java/org/ethereum/datasource/KeyValueDataSourceUtils.java

```diff
@@ -98,8 +98,9 @@ public static void validateDbKind(DbKind currentDbKind, String databaseDir, bool
         }
 
         boolean databaseDirExists = dir.exists() && dir.isDirectory();
+        File file = new File(databaseDir, DB_KIND_PROPERTIES_FILE);
 
-        if (!databaseDirExists || dir.list().length == 0) {
+        if (!databaseDirExists || dir.list().length == 0 || !file.exists() || !file.canRead()) {
             KeyValueDataSourceUtils.generatedDbKindFile(currentDbKind, databaseDir);
```

Comment:
If the file does exists but can't be read for some reason, the code will default to rocks db if we don't do that check, instead of generating the file with the correct DB.